js/NavMainComponent.js

```diff
@@ -126,18 +129,39 @@ NavMainComponent.prototype.init = function () {
     this.$html.on('keydown', function (event) {
         if (event.keyCode === 27) {
             if (component.$navQuaternary.hasClass('is-open')) {
-                component.closeQuaternaryNav();
+                component.closeQuaternaryNav({ type: NavMainComponent.CLOSE_WITH_ESCAPE });
             } else if (component.$navTertiary.hasClass('is-open')) {
                 component.closeTertiaryNav();
             } else if (component.$navSecondary.hasClass('is-open')) {
-                component.closeSecondaryNav();
+                component.closeSecondaryNav({ type: NavMainComponent.CLOSE_WITH_ESCAPE });
             } else if (isMobile && component.$body.hasClass('open-nav')) {
                 component.showMobileNav(false)
             }
         }
     });
 }
 
+/**
+ * Check if any navs are open
+ * @returns {boolean}
+ */
+NavMainComponent.prototype.isNavOpen = function () {
+    var component = this,
+        isMobile = !component.window.matchMedia('(min-width: 992px)').matches;
+
+    if (component.$navQuaternary.hasClass('is-open')) {
```

Comment:
It's a minor thing, but seems like you've ordered this from least-likely to most-likely, flipping it would mean we could get out of this conditional sooner in most cases

Reply:
Done